src/queue/bookmark/workflows.ts

```diff
@@ -0,0 +1,28 @@
+import { proxyActivities, sleep } from '@temporalio/workflow';
+import { BookmarkActivities } from './activities';
+
+export interface BookmarkReminderParams {
+  userId: string;
+  postId: string;
+  remindAt: number;
+}
+
+export async function bookmarkReminderWorkflow({
+  userId,
+  postId,
+  remindAt,
+}: BookmarkReminderParams): Promise<void> {
+  const { validateBookmark, sendBookmarkReminder } =
+    proxyActivities<BookmarkActivities>({ scheduleToCloseTimeout: '15s' }); // the amount of time the process is willing to wait for the activity to complete
+
+  const isValid = await validateBookmark({ userId, postId });
+
+  if (!isValid) {
+    return;
+  }
+
+  const diff = remindAt - Date.now();
+
+  await sleep(diff);
```

Comment:
I'm not sure but potentially there's a function that accepts date and triggers automatically. Can you check please instead of using sleep?

Reply:
Yeah, sure I can check for alternatives. I will get back to this.

Reply:
The `sleep` method is told to be the [way to go](https://community.temporal.io/t/delayed-execution-of-workflow/3390/2), though the answer from the moderator was 3 years ago. Then a year after that there is an introduction of `delay` as part of the `WorkflowOptions`. I can maybe ask them for an updated answer.

Reply:
[They replied](https://community.temporal.io/t/sleep-vs-start-delay/12877/2?u=sshanzel) and it seems a start delay is being recommended. I shall update the process.